cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -333,20 +332,18 @@ public boolean apply(ArtifactDetail artifactDetail) {
    * @param artifactId the id of the artifact to get plugins for
    * @param pluginType the type of plugins to get
    * @param pluginName the name of plugins to get
-   * @param pluginArtifactRange the range of the artifacts the plugin is in, null for all artifacts
-   * @param limit the limit number of the result, Integer.MAX_VALUE means return all results
+   * @param pluginPredicate the predicate for the plugin
+   * @param limit the limit number of the result
    * @param order the order of the result
    * @return an unmodifiable sorted map from plugin artifact to plugins in that artifact
    * @throws ArtifactNotFoundException if the given artifact does not exist
    * @throws IOException if there was an exception reading plugin metadata from the artifact store
    */
-  public SortedMap<ArtifactDescriptor, PluginClass> getPlugins(NamespaceId namespace, Id.Artifact artifactId,
-                                                               String pluginType, String pluginName,
-                                                               @Nullable ArtifactVersionRange pluginArtifactRange,
-                                                               int limit, ArtifactSortOrder order)
-    throws IOException, PluginNotExistsException, ArtifactNotFoundException {
-    return artifactStore.getPluginClasses(namespace, artifactId, pluginType, pluginName,
-                                          pluginArtifactRange, limit, order);
+  public SortedMap<ArtifactDescriptor, PluginClass> getPlugins(
+    NamespaceId namespace, Id.Artifact artifactId, String pluginType, String pluginName,
+    com.google.common.base.Predicate<co.cask.cdap.proto.id.ArtifactId> pluginPredicate,
```

Comment:
This is basically because in `ArtifactStore`, we are using google's Predicate, and in `ArtifactRepository` we are using cdap's Predicate, I will make it consistent.